diff --git a/src/isa/mod.rs b/src/isa/mod.rs
index 8728c29..6dd6910 100644
--- a/src/isa/mod.rs
+++ b/src/isa/mod.rs
@@ -420,7 +420,7 @@ pub trait TargetIsa: fmt::Display + Send + Sync {
 
 /// A wrapper around the ISA-dependent flags types which only implements `Hash`.
 #[derive(Hash)]
-pub struct IsaFlagsHashKey<'a>(&'a [u8]);
+pub struct IsaFlagsHashKey<'a>(pub &'a [u8]);
 
 /// Function alignment specifications as required by an ISA, returned by
 /// [`TargetIsa::function_alignment`].
diff --git a/src/lib.rs b/src/lib.rs
index 64d37df..77ed7a9 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,5 +1,5 @@
 //! Cranelift code generation library.
-#![deny(missing_docs)]
+#![allow(hidden_glob_reexports, missing_docs)]
 // Display feature requirements in the documentation when building on docs.rs
 #![cfg_attr(docsrs, feature(doc_cfg))]
 #![no_std]
@@ -41,7 +41,7 @@ pub use gimli;
 include!(concat!(env!("ISLE_DIR"), "/isle_numerics.rs"));
 
 #[macro_use]
-mod machinst;
+pub mod machinst;
 
 pub mod binemit;
 pub mod cfg_printer;
